### PR TITLE
Improve article factory options

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Rather than manipulating the result, this adds more options to the article factory used in the tests.

Refs https://github.com/libero/article-store/pull/115#discussion_r358286937.

### CODE REVIEW

File: test/create-article.ts [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
 import { termToString } from 'rdf-string';
 
 const isNotUndefined = (arg: unknown): boolean => arg !== undefined;
 
-export default (id?: BlankNode, name = id ? `Article ${id}` : 'Article'): JsonLdObj => (
+type Options = {
+  id?: BlankNode;
+  name?: string;
+  types?: Array<NamedNode>;
+}
```
Comment (thewilkybarkid) [inline test/create-article.ts:14]: Using an object to avoid argument proliferation.

File: test/create-article.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 import { JsonLdObj } from 'jsonld/jsonld-spec';
-import { BlankNode } from 'rdf-js';
+import { BlankNode, NamedNode } from 'rdf-js';
 import { termToString } from 'rdf-string';
 
 const isNotUndefined = (arg: unknown): boolean => arg !== undefined;
 
-export default (id?: BlankNode, name = id ? `Article ${id}` : 'Article'): JsonLdObj => (
+type Options = {
+  id?: BlankNode;
```
Comment (giorgiosironi) [inline test/create-article.ts:11]: I guess it doesn't make things easier to pass in a `string` `id`, as we need the `BlankNode` in the other calls to `set(...)` anyway.

File: test/create-article.ts [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
+
+export default ({
+  id = undefined,
+  name = 'Article',
+  types = [namedNode('http://schema.org/Article')],
+}: Options = {}): JsonLdObj => (
   deepFilter({
     '@id': id ? termToString(id) : undefined,
-    '@type': 'http://schema.org/Article',
+    '@type': types.map(termToString),
```
Comment (giorgiosironi) [inline test/create-article.ts:23]: why passing in `NamedNode` instances if they have to be cast down to string anyway? Any validation happening in `namedNode()`?

Comment (thewilkybarkid) [reply test/create-article.ts:23]: Yep, it needs to be an IRI.

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**